sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -56,17 +61,24 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
 
   def acquire(numPermits: Int, priority: T): Unit = {
     lock.lock()
+    val condition = lock.newCondition()
+    val info = ThreadInfo(priority, condition, numPermits)
     try {
-      val condition = lock.newCondition()
-      while (!canAcquire(numPermits)) {
-        waitingQueue.enqueue(ThreadInfo(priority, condition))
-        condition.await()
+      if (canAcquire(numPermits)) {
+        commitAcquire(numPermits)
+      } else {
+        waitingQueue.enqueue(info)
+        while (!info.signaled) {
+          info.condition.await()
+        }
       }
-      commitAcquire(numPermits)
-
+    } catch {
+      case _: InterruptedException =>
+        info.interrupted = true
```

Comment:
This eats the InterruptedException and does not rethrow it elsewhere.  IMO we don't want to special-case InterruptedException here.  Instead I think we want to handle any exception escaping here and remove ourselves from the queue before rethrowing the exception.

Reply:
Handling all exceptions and rethrowing makes sense to me. Regarding removing from the queue, are you ok with this pattern of invalidating the entry instead of explicitly removing it? I wasn't able to find a good way to remove a specific element from the queue without having to rebuild the entire queue. I suppose I could also try using our `HashedPriorityQueue` for that.

Reply:
I am fine if you invalidate it.

Reply:
Why not use java.util.PriorityQueue which does allow this?